modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/addimages/plugin.js

```diff
@@ -167,6 +219,58 @@
 					this._processFile(imageFile, event.editor);
 				}
 			}
+			else if (event.data && event.data.type === 'html') {
+				const editor = event.editor;
+				const fragment = CKEDITOR.htmlParser.fragment.fromHtml(
+					event.data.dataValue
+				);
+
+				fragment.forEach((element) => {
+					if (
+						element.type === CKEDITOR.NODE_ELEMENT &&
+						element.name === 'img'
+					) {
+						const base64Regex = /^data:image\/(.*);base64,/;
+						const match = element.attributes.src.match(base64Regex);
+
+						if (element.attributes.src && match) {
+							const src = element.attributes.src;
+							const extension = match[1];
+							const name = `${Date.now().toString()}.${extension}`;
+
+							// Mark this "src" attribute as something we want to replace
+
+							this._imageSrcToReplace.push(src);
```

Comment:
`this._tempImage` is an image and here `src` is a string. There's definitely room for improvement but you can't use the same logic for different use cases. When you drag and drop an image from your desktop you get a File, when you paste from a document you don't

Reply:
Both cases converge on `imageAdd` handler, where we have image node reference, so it can be used in the same way. Why couldn't we use it in your case?

Let me backtrack a bit on the array part of https://github.com/liferay-frontend/liferay-portal/pull/225#discussion_r468424021, I misundertood that we are trying to support multiple images. I believe it's not a goal to support it in this PR.

Reply:
Yeah you're right, we can reuse `this._tempImage` and get the same result. Thanks